app/src/common/shared/org/mozilla/vrbrowser/browser/engine/Session.java

```diff
@@ -521,19 +517,24 @@ private GeckoSession createGeckoSession(@NonNull SessionSettings aSettings) {
             session = new GeckoSession(geckoSettings);
         }
 
-        session.getSettings().setUserAgentOverride(aSettings.getUserAgentOverride());
+        if (session != null) {
+            session.getSettings().setUserAgentOverride(aSettings.getUserAgentOverride());
+        }
         setupSessionListeners(session);
 
         return session;
     }
 
-    public void recreateSession() {
+    void recreateSession() {
         boolean wasFullScreen = mState.mFullScreen;
 
         SessionState previous = mState;
         mState = mState.recreate();
 
-        restore();
+        mSessionChangeListeners.forEach(listener -> {
+            listener.onSessionClosed(this);
+            listener.onSessionRemoved(this);
+        });
```

Comment:
Why are we closing the session and removing it after it is recreated?

Reply:
Yep, good catch, we don't need to do that, just close the old one and open the new one. I've updated.